src/driver/single-connection-provider.ts

```diff
@@ -17,16 +17,11 @@ export class SingleConnectionProvider implements ConnectionProvider {
     }
 
     const promise = this.#run(consumer)
-
     this.#runningPromise = promise
-      .then(() => {
-        this.#runningPromise = undefined
-      })
-      .catch(() => {
-        this.#runningPromise = undefined
-      })
 
-    return promise
+    return promise.finally(() => {
+      this.#runningPromise = undefined
+    })
```

Comment:
Doesn't this break the whole logic? `this.#runningPromise` now doesn't have the finally block. When you await `this.#runningPromise`, it won't be set to undefined once the promise resolves. The while loop above will go to a busy loop for a while.

Reply:
It probably works, but this may have destroyed the performance.

Reply:
Another issue. The promise stored to `this.#runningPromise` will now throw and the second query waiting for a connection will also fail.